src/core/crypto/util/misc.h

```diff
@@ -0,0 +1,57 @@
+ * Redistribution and use in source and binary forms, with or without modification, are       //
+ * permitted provided that the following conditions are met:                                  //
+ *                                                                                            //
+ * 1. Redistributions of source code must retain the above copyright notice, this list of     //
+ *    conditions and the following disclaimer.                                                //
+ *                                                                                            //
+ * 2. Redistributions in binary form must reproduce the above copyright notice, this list     //
+ *    of conditions and the following disclaimer in the documentation and/or other            //
+ *    materials provided with the distribution.                                               //
+ *                                                                                            //
+ * 3. Neither the name of the copyright holder nor the names of its contributors may be       //
+ *    used to endorse or promote products derived from this software without specific         //
+ *    prior written permission.                                                               //
+ *                                                                                            //
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY        //
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF    //
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL     //
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,       //
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,               //
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS    //
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,          //
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF    //
+ * THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.               //
+ */
+
+#ifndef SRC_CORE_CRYPTO_UTIL_MISC_H_
+#define SRC_CORE_CRYPTO_UTIL_MISC_H_
+
+#include <cryptopp/misc.h>
+
+namespace kovri
+{
+namespace core
+{
+/// @brief Constant-time comparison of two equally sized buffers
+/// @param buf1 First buffer
+/// @param buf2 Second buffer
+/// @param size Byte size of both buffers
+/// @notes From Crypto++:
+///   "The function effectively performs an XOR of the elements in two equally sized buffers
+///   and retruns a result based on the XOR operation. The function is near constant-time because
+///   CPU micro-code timings could affect the "constant-ness". Calling code is responsible for
+///   mitigating timing attacks if the buffers are not equally sized."
+bool ConstTimeCmp(
```

Comment:
I aware of the memcmp function..... I'm simply pointing out the the API defined by Crypto++ is written as VerifyBufsEqual which helps the caller understand that a return of true means they are equal.

By adjusting the interface to ConstTimeCmp it adds ambiguity on whether or not it acts more like the return of memcmp or not. Regardless of your view on using the return of memcmp as a boolean, this was/is what the existing code does.

As a new reader of the code who just sees ConstTimeCmp, I have to look into the implementation of the function  and possible the implementation of VerifyBufsEqual to understand how to use it. And, it would be easier to just read the docstring and understand what it does.

Maybe that is OK or maybe it isn't, but telling me to rtfm doesn't improve the discussion.

Reply:
>Maybe that is OK or maybe it isn't, but telling me to rtfm doesn't improve the discussion.

Neither does that comment.

>By adjusting the interface to ConstTimeCmp it adds ambiguity on whether or not it acts more like the return of memcmp or not. 

If you had actually read the gitlog, and the TODOs in the commit, and the issue noted in both (#895), you'll see that this implementation will change in the near future. If you really need `@return True if buffers are equal` in the meantime, then ok, that can be added. Or, you could've added it yourself in a PR to my fork in this very branch (github facilitates that very easily).

>And, it would be easier to just read the docstring and understand what it does.

Don't trust docs. Trust the code.